From: Shimazu <vanessimbolon2020@gmail.com>
Date: Fri, 31 Jul 2026 02:19:04 +0000
Subject: BoostEffect: Rename variables and migrate to EntityScheduler

AE PATCH REASON: Obfuscated variable names hinder maintainability; Bukkit.getScheduler() incompatible with Folia.
AE PATCH FIX: Rename var1/var2/var3 to meaningful names throughout. Migrate Bukkit.getScheduler().runTaskLater() to target.getScheduler().runDelayed().
AE PATCH RISK: LOW

---
diff --git a/net/advancedplugins/ae/impl/effects/effects/effects/internal/BoostEffect.java b/net/advancedplugins/ae/impl/effects/effects/effects/internal/BoostEffect.java
--- a/net/advancedplugins/ae/impl/effects/effects/effects/internal/BoostEffect.java	2026-07-30 10:27:31.664350087 +0000
+++ b/net/advancedplugins/ae/impl/effects/effects/effects/internal/BoostEffect.java	2026-07-30 10:28:05.857658347 +0000
@@ -21,61 +21,58 @@
 public class BoostEffect extends AdvancedEffect {
    private final List<UUID> ignoreFall = new ArrayList<>();
 
-   public BoostEffect(JavaPlugin var1) {
-      super(var1, "BOOST", "Boost entity up in air", "%e:[DIRECTION]:[AMOUNT]");
+   public BoostEffect(JavaPlugin plugin) {
+      super(plugin, "BOOST", "Boost entity up in air", "%e:[DIRECTION]:[AMOUNT]");
       this.addArgument(0, Integer.class);
    }
 
    @Override
-   public boolean executeEffect(ExecutionTask var1, LivingEntity var2, String[] var3) {
-      Bukkit.getScheduler()
-         .runTaskLater(
-            EffectsHandler.getInstance(),
-            () -> {
+   public boolean executeEffect(ExecutionTask task, LivingEntity target, String[] args) {
+      target.getScheduler().runDelayed(ASManager.getInstance(), t -> {
                try {
-                  String var3x = "UP";
-                  double var4 = 1.0;
-                  if (var3.length == 1) {
-                     if (MathUtils.isDouble(var3[0])) {
-                        var4 = Double.parseDouble(var3[0]);
+                  String dir = "UP";
+                  double amount = 1.0;
+                  if (args.length == 1) {
+                     if (MathUtils.isDouble(args[0])) {
+                        amount = Double.parseDouble(args[0]);
                      } else {
-                        var3x = var3[0];
+                        dir = args[0];
                      }
-                  } else if (var3.length != 0) {
-                     var3x = var3[0];
-                     if (!MathUtils.isDouble(var3[1])) {
-                        EffectsHandler.getInstance().getLogger().severe("'BOOST' effect was used with an invalid amount! Expected a number, got " + var3[1]);
+                  } else if (args.length != 0) {
+                     dir = args[0];
+                     if (!MathUtils.isDouble(args[1])) {
+                        EffectsHandler.getInstance().getLogger().severe("'BOOST' effect was used with an invalid amount! Expected a number, got " + args[1]);
                         return;
                      }
 
-                     var4 = MathUtils.clamp(Double.parseDouble(var3[1]), -20.0, 20.0);
+                     amount = MathUtils.clamp(Double.parseDouble(args[1]), -20.0, 20.0);
                   }
 
-                  Vector var6 = var2.getLocation().getDirection().normalize();
-                  double var7 = var4 / 10.0;
-                  double var9 = var6.getX() * var7;
-                  double var11 = var6.getZ() * var7;
-                  String var14 = var3x.toUpperCase(Locale.ROOT);
-                  Vector var13;
-                  switch (var14) {
+                  Vector direction = target.getLocation().getDirection().normalize();
+                  double number = amount / 10.0;
+                  double x = direction.getX() * number;
+                  double z = direction.getZ() * number;
+                  String directionUpper = dir.toUpperCase(Locale.ROOT);
+                  Vector boost;
+                  switch (directionUpper) {
                      case "UP":
-                        var13 = new Vector(0.0, var7, 0.0);
+                        boost = new Vector(0.0, number, 0.0);
                         break;
                      case "DOWN":
-                        var13 = new Vector(0.0, -var7, 0.0);
+                        boost = new Vector(0.0, -number, 0.0);
                         break;
                      case "FORWARD":
-                        var13 = new Vector(var9, 0.0, var11);
+                        boost = new Vector(x, 0.0, z);
                         break;
                      case "BACKWARD":
-                        var13 = new Vector(-var9, 0.0, -var11);
+                        boost = new Vector(-x, 0.0, -z);
                         break;
                      case "LOOK":
-                        Vector var16 = var2.getEyeLocation().getDirection().normalize();
-                        double var17 = var16.getX() * var7;
-                        double var19 = var16.getZ() * var7;
-                        double var21 = var16.getY() * var7;
-                        var13 = new Vector(var17, var21, var19);
+                        Vector eyeDirection = target.getEyeLocation().getDirection().normalize();
+                        double eX = eyeDirection.getX() * number;
+                        double eZ = eyeDirection.getZ() * number;
+                        double eY = eyeDirection.getY() * number;
+                        boost = new Vector(eX, eY, eZ);
                         break;
                      default:
                         EffectsHandler.getInstance()
@@ -84,32 +81,30 @@
                         return;
                   }
 
-                  if (ASManager.isExcessVelocity(var13)) {
+                  if (ASManager.isExcessVelocity(boost)) {
                      EffectsHandler.getInstance().getLogger().severe("'BOOST' effect used with too much velocity! Please lower the effect value.");
                      return;
                   }
 
-                  var2.setVelocity(var13);
-                  if (!this.ignoreFall.contains(var2.getUniqueId())) {
-                     this.ignoreFall.add(var2.getUniqueId());
+                  target.setVelocity(boost);
+                  if (!this.ignoreFall.contains(target.getUniqueId())) {
+                     this.ignoreFall.add(target.getUniqueId());
                   }
-               } catch (Exception var23) {
-                  var23.printStackTrace();
+               } catch (Exception e) {
+                  e.printStackTrace();
                }
-            },
-            1L
-         );
+            }, null, 1L);
       return true;
    }
 
    @EventHandler
-   public void onFall(EntityDamageEvent var1) {
-      if (var1.getEntity().getType().equals(EntityType.PLAYER)) {
-         if (var1.getCause().equals(DamageCause.FALL)) {
-            UUID var2 = var1.getEntity().getUniqueId();
-            if (this.ignoreFall.contains(var2)) {
-               var1.setCancelled(true);
-               this.ignoreFall.remove(var2);
+   public void onFall(EntityDamageEvent e) {
+      if (e.getEntity().getType().equals(EntityType.PLAYER)) {
+         if (e.getCause().equals(DamageCause.FALL)) {
+            UUID u = e.getEntity().getUniqueId();
+            if (this.ignoreFall.contains(u)) {
+               e.setCancelled(true);
+               this.ignoreFall.remove(u);
             }
          }
       }
